### ISSUE TITLE
Spelling Practice Auto-advance

### ISSUE THREAD

**Issue by ldv8434** [SethClydesdale/genki-study-resources#196]
# Issue
I'm encountering an issue with my IME and what I'm guessing is an auto-advance function on the text boxes in "Spelling Practice" vocab exercises. 

When I go to convert to kanji and highlight the correct one (which changes the contents of the text box to the selected kanji), the website will immediately send me to the next text box. Unfortunately, it also copies the same kanji to the textbox it moved to. In effect, I only need to type the word 3 times to complete the row instead of the expected six unless I go back and manually erase them.

## Context
I'm using FCITX5 + Mozc + KDE Plasma + Arch Linux. Mozc is set to input as Romaji, auto-convert to kana as they're completed, and start converting to kanji at the press of space or tab. 

# Visual example
This is Lesson 11: Nouns 2. The word is entered and the correct answer is about to be selected. 
![image](https://user-images.githubusercontent.com/65482502/220262859-67e58466-2f02-4885-9b46-93b88a2e2479.png)
Here the result after pressing tab:
![image](https://user-images.githubusercontent.com/65482502/220262991-cfdcde1c-c0e7-432c-98fc-53111fa4721d.png)

# Expected Behavior
Boxes only move forward by one at a time when inputting the answer. Though, I expect this is an issue moreso at the IME level and not something you should be expected to fix. Can we get a toggle in the settings to disable the auto-advance on spelling practice exercises? 

---

As an unrelated site note: thank you so much for this site! It makes checking answers so much easier.

**Comment by SethClydesdale** [comment]
Hello @ldv8434, thank you for reporting this! I've attempted some fixes for this in the past that _have_ worked, however, it seems that it's still persisting on some IMEs. I'll definitely look into adding an option to disable the auto-advancement, and possibly a notice about this bug in the information portion.

**Comment by SethClydesdale** [comment]
Hello again @ldv8434!

After reviewing the auto-checker/advancement function, I noticed that it wasn't working as it should have been (properly queueing the next check and clearing the duplicated value). I made some adjustments and it *should* be working properly now, hopefully. Do let me know if the problem is still persisting! I would recommend clearing your cache first before testing, since the old function may be preserved.
